hal/targets/hal/TARGET_STM/TARGET_STM32L4/sleep.c

```diff
@@ -46,13 +46,27 @@ void deepsleep(void)
 {
     // Stop HAL systick
     HAL_SuspendTick();
-
-    // Request to enter STOP mode with regulator in low power mode
-    HAL_PWR_EnterSTOPMode(PWR_LOWPOWERREGULATOR_ON, PWR_STOPENTRY_WFI);
+  
+    if(__HAL_RCC_PWR_IS_CLK_ENABLED())
```

Comment:
There is also a deepsleep wakeup delay problem, I believe it needs 0.3 secs instead of 10 usecs. Using the regular sleep there is no wakeup delay. Maybe this is related to this problem. Regards Helmut